### DISCUSSION TITLE
why mpv takes 15% GPU usage for just a 720p video playing?

### DISCUSSION THREAD

**Discussion by jcyuan** [mpv-player/mpv#14161] [Q&A] [answered]
os: win10
mpv: 0.38.0
graphics: opengles ANGLE
sample video: https://docs.evostream.com/sample_content/assets/sintel1m720p.mp4

i don't think this is normal? but why?

**Comment by jcyuan** [comment]
btw: i'm using c# binding with mpv

**Comment by llyyr** [comment]
`--profile=fast` should help. Hard to say more without seeing logs or your specs. 15% gpu usage is pretty high for a modern gpu, pretty low for a 15 year old toaster

**Reply by jcyuan** [reply]
😅 yeah thanks for reply and I'm sorry about that.
my gpu is geforce 1080, enough i think.
but not sure if the problem was caused by the gui framework or not, I'll dig into the problem a little deeper and go back to you. 
about "specs" i assume you mean the config i use to play? if so, i didn't set a config, it was just a very basic and quick playing test.

**Reply by Raza-0007** [reply]
Geforce 1080 should not have any issues playing any video even with high-quality profile.

Don't be too concerned with a utilization percentage as Nvidia GPU's vary their graphics clock frequency according to the load. So the GPU running at 700 MHz, playing the same file will show almost double utilization percentage when compared to the same GPU running at 1400 MHz. A better measurement is the GPU power consumption which will remain the same for the above two cases. The GPU itself decides what clock frequency to select. You can see it yourself by using CPUID HWMonitor while playing a video.

~Raza

**Reply by jcyuan** [reply, marked as answer]
sorry guys, it was caused by the GUI framework, the swapchain has bug.
the real GPU usage on my PC was about 5% - 6%, it's good.
